### PR TITLE
Allow users to specify Azure tags

### LINKED ISSUE #1966: [ENH] - Allow Azure users to specify tags via the nebari-config.yaml
### Feature description

Azure resources can have tags applied for the purposes of tracking and organizing resources. We have tags for other cloud providers but not for Azure.

### Value and/or benefit

If an organization has a tag policy, this will allow those deploying Nebari to easily adhere to those policies from the moment the Azure resources are deployed.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs
Closes #1966 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: src/_nebari/stages/terraform_state/__init__.py [hunk 1]
```diff
@@ -56,6 +57,30 @@ def _validate_resource_group_name(cls, value):
...
+                    f"Tag name '{tag_name}' exceeds maximum length of {max_name_length} characters."
+                )
+
+            if len(tag_value) > max_value_length:
+                raise ValueError(
+                    f"Tag value '{tag_value}' for tag '{tag_name}' exceeds maximum length of {max_value_length} characters."
+                )
+
+        return tags
+
```
Comment (Adam-D-Lewis) [inline src/_nebari/stages/terraform_state/__init__.py:83]: Why do we have to define this function twice?  Can't we import in one place or the other?

**Comment by Adam-D-Lewis** [comment]
LGTM assuming it deploys.  Does azurerm_kubernetes_cluster pass the tags onto the load balancer as well?

**Comment by iameskild** [comment]
@Adam-D-Lewis I ended up including `max_pods` and `network_profile` as variables for Azure as well. These were added to ensure we can deploy into a private cluster and gives us a lot more flexibility during our deployments. Free feel to give this another quick review, thanks :)

**Comment by Adam-D-Lewis** [comment]
LGTM!

**Pull request merged**